osu.Game/Rulesets/Objects/SliderPath.cs

```diff
@@ -262,9 +262,16 @@ private void calculatePath()
                 var segmentVertices = vertices.AsSpan().Slice(start, i - start + 1);
                 var segmentType = ControlPoints[start].Type ?? PathType.Linear;
 
-                foreach (Vector2 t in calculateSubPath(segmentVertices, segmentType))
+                // No need to calculate path when there is only 1 vertex
+                if (segmentVertices.Length == 1)
+                    calculatedPath.Add(segmentVertices[0]);
```

Comment:
This appears to be a mostly unrelated optimisation, but I can't see it being wrong and sheets are silend, so I guess I'll let it be...